### ISSUE TITLE
Invalid Header Length Error

### ISSUE THREAD

**Issue by lordzouga** [mike01/pypacker#11]
Thanks for this library, it has certainly made my life easier! but I have this little problem..

I captured an IP Packet on an android device and tried to parse TCP and I got a invalid header length error.
here is the packet's byte string:
```
b'E\x00\x00<F\x84@\x00@\x06\xa0)\n\x00\x02\x00.\x04\x1a\x0b\xcd\x9a\x00P\ri\x83\x16\x00\x00\x00\x00\xa0\x02|8Sa\x00\x00\x02\x04\x02\x12\x04\x02\x08\n\x01<\xc8T\x00\x00\x00\x00\x01\x03\x03\x06''
```
and here is the code I use to parse it 
``` python
from pypacker.layer3 import ip
from pypacker.layer4 import tcp

pbytes = #bytestring above
ipp = ip.IP(pbytes)
if ipp[tcp.TCP] is not None:
        print(tcp.TCP(pbytes))
```
Whenever, It gets to the "print(tcp.TCP(body))" line, it gives this error:
```
ERROR (__init__): could not dissect, trying to unpack after all..
Traceback (most recent call last):
  File "C:\Python33\lib\site-packages\pypacker\pypacker.py", line 257, in __init__
    self._dissect(args[0])
  File "C:\Python33\lib\site-packages\pypacker\layer4\tcp.py", line 142, in _dissect
    raise pypacker.UnpackError("invalid header length")
```

Please, can you help me fix this?

**Comment by mike01** [comment]
Hmm I'm not sure what your goal is like..printing the tcp bytes if ipp contains a TCP layer?  "ipp" doesn't seem to be used after initiation. In order to print the tcp bytes you could do one of the following:
``` python
pack = ip.IP(pbytes)

if pack[tcp.TCP] is not None:
        # print tcp structure
        print(pack.tcp)
        # print tcp raw bytes (header+body)
        print(pack.tcp.bin())
        # print tcp raw bytes (header)
        print(pack.tcp.header_bytes)
        # print tcp raw bytes (body)
        print(pack.tcp.body_bytes)
```

**Comment by lordzouga** [comment]
Thanks for your reply.
The method you gave certainly works better, The parsed result from your library now corresponds with the one I got from wireshark.

The goal for me is to verify that the packet I generated was not corrupt and the library gives accurate results every time.
I have edited the code to properly illustrate what I tried. I have no idea what made write it the way I did previously.

Thanks again! closing.
